src/middleware/firmware_update_usb.rs

```diff
@@ -15,20 +17,22 @@ pub trait FwUpdate: AsyncRead + AsyncWrite + AsyncSeek + std::marker::Unpin + Se
 pub type FactoryItem = BoxFuture<'static, Result<Box<dyn FwUpdate>, FlashingError>>;
 
 pub fn fw_update_factory(
-    vid_pid: (u16, u16),
+    device: rusb::Device<GlobalContext>,
```

Comment:
Shouldn't this (and subsequently `Rk1FwUpdateDriver::new()` take `device` by reference?

Reply:
Could you elaborate?

Reply:
`Rk1FwUpdateDriver::new()` does not consume `device`, therefore we can avoid unnecessary copying

Reply:
it needs to be available in the async context. Passing it by reference would mean the future would outlive the borrow of `Device`. What i can do is delay the clone, then i would not get constructed on default in the rpi case